sky/utils/ux_utils.py

```diff
@@ -121,6 +123,81 @@ def run(self, *args, **kwargs):
                 raise
 
 
+class RedirectOutputForThread:
+    """Redirects stdout and stderr to thread-specific files.
+
+    This class enables output redirection for threading.Thread.
+    Example usage:
+    with RedirectOutputForThread() as redirector:
+        with ThreadPoolExecutor() as executor:
+            executor.submit(redirector.wrap(task_func, 'task1.log'), *args)
+    """
+
+    def __init__(self) -> None:
+        self._original_stdout: TextIO = sys.stdout
+        self._original_stderr: TextIO = sys.stderr
+
+    class _ThreadAwareOutput:
+        """A thread-aware output class that replaces sys.stdout/stderr."""
+
+        def __init__(self, original_output: TextIO) -> None:
+            self._original_output: TextIO = original_output
+            self._thread_local: Optional[threading.local] = None
+
+        def resolve_output(self) -> TextIO:
+            if self._thread_local and hasattr(self._thread_local, 'file'):
+                return typing.cast(TextIO, self._thread_local.file)
+            return self._original_output
+
+        def write(self, data: str):
+            output = self.resolve_output()
+            output.write(data)
+
+        def flush(self):
+            output = self.resolve_output()
+            output.flush()
+
+        def set_thread_local(self, thread_local: Optional[threading.local]):
+            self._thread_local = thread_local
+
+    def __enter__(self):
+        """Replaces global stdout/stderr with our thread-aware versions."""
+        self._thread_aware_stdout, self._thread_aware_stderr = (
+            self._ThreadAwareOutput(self._original_stdout),
+            self._ThreadAwareOutput(self._original_stderr))
+        sys.stdout, sys.stderr = (self._thread_aware_stdout,
+                                  self._thread_aware_stderr)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Restores original stdout/stderr."""
+        sys.stdout, sys.stderr = (self._thread_aware_stdout,
+                                  self._thread_aware_stderr)
+
+    def wrap(self, func: Callable, filepath: str, mode: str = 'w'):
```

Comment:
```suggestion
    def run(self, func: Callable, filepath: str, mode: str = 'w'):
```
Lets keep align with RedirectOutputForProcess?

Reply:
While I think this is better...